crates/jit/src/link.rs

```diff
@@ -142,12 +142,17 @@ cfg_if::cfg_if! {
             pub fn ___chkstk();
         }
         const PROBESTACK: unsafe extern "C" fn() = ___chkstk;
+    } else if #[cfg(not(any(target_arch = "x86_64", target_arch = "x86")))] {
+        // As per
+        // https://github.com/rust-lang/compiler-builtins/blob/cae3e6ea23739166504f9f9fb50ec070097979d4/src/probestack.rs#L39,
+        // LLVM only has stack-probe support on x86-64 and x86. Thus, on any other CPU
+        // architecture, we simply use an empty stack-probe function.
+        extern "C" fn empty_probestack() {}
```

Comment:
The `probestack` has an ABI that is not compatible with `C`. This means that this will only work correctly if the compiler happened to decide that only a `ret` should be emitted. If it does anything else, like writing to a caller saved register that was used as argument for the caller of `probestack`, you will likely get a crash.

Reply:
Yes, that's a theoretical concern. In practice, the empty function `{}` compiles to just a `ret` in both debug and release mode on both x86-64 and aarch64 (just tested all four combinations). Do you have a suggestion for a better approach (compiler attributes, etc)?

Reply:
Maybe use Cranelift to create the function?

Reply:
That seems like quite a bit of overhead, and potentially fragile and complex code to maintain (a new "probestack" ABI to implement on ARM64, a codegen step at startup, a way to lazily initialize the `PROBESTACK` global, etc.) for a theoretical concern that does not actually occur in practice. I'll defer to @sunfishcode 's judgment here, but my inclination is to leave it as-is.

Reply:
As it turns out, @alexcrichton's https://github.com/bytecodealliance/wasmtime/pull/1490 will eliminate the need for this call entirely, which avoids this question entirely.

Beyond that, if we want to reintroduce stack probes, we should add a `__rust_probestack` for AArch64 and any other platform we need it on [here](https://github.com/rust-lang/compiler-builtins/blob/master/src/probestack.rs) rather than having our own version in Cranelift.

Reply:
@sunfishcode, that sounds perfect. Would you like us to wait for #1490 to land, or would it be acceptable to land this with the temporary "empty function" stack probe in the meantime (if the timing works out that way)?

Reply:
I think it's fine to land it with this temporary workaround.